misc/idp-sync/api.md

```diff
@@ -0,0 +1,335 @@
+  --header 'Authorization: Token <PAT>' \
+  --header 'Content-Type: application/json' \
+  --data '{
+	"serviceAccountKey": "<SERVICE_ACCOUNT_KEY>",
+	"customerID": "<CUSTOMER_ID>"
+}'
+```
+
+Response
+```json
+{
+	"id": <ID>,
+	"customerId": "<CUSTOMER_ID",
+	"syncInterval": 300,
+	"enabled": true
+}
+```
+
+### Get Integration by ID
+Request
+```shell
+curl --request GET \
+  --url https://api.netbird.io/api/integrations/google-idp/<ID> \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <PAT>'
+```
+
+Response
+```json
+{
+  "id": <ID>,
+  "customerId": "<CUSTOMER_ID",
+  "syncInterval": 300,
+  "enabled": true
+}
+```
+
+### Get All Integrations By AccountID
+Request
+```shell
+curl --request GET \
+  --url https://api.netbird.io/api/integrations/google-idp \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <PAT>'
+```
+
+Response
+```json
+[
+  {
+    "id": <ID>,
+    "customerId": "<CUSTOMER_ID>",
+    "syncInterval": 300,
+    "enabled": true
+  }
+]
+```
+
+### Force Integration Sync
+Request
+```shell
+curl --request GET \
+  --url https://api.netbird.io/api/integrations/google-idp/<ID>/sync \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <PAT>'
+```
+
+Response
+```json
+{
+  "result": "ok"
+}
+```
+
+### Update Integration
+Updates the selected parameters for a specific integration.
+
+Request
+- `serviceAccountKey`: A Base64 encoded string derived from a service account key JSON.For the creation of the service account key JSON, refer to the provided [IdP guideline](idp.md).
+  Encode service account JSON to base64 by using the command:
+```shell
+  base64 -i <SERVICE_ACCOUNT_KEY_PATH>
+``` 
+-  `syncInterval`: Optional. Should not be less than 300 seconds.
+- `enabled`: Optional. Used to disable/enable the integration. 
+
+```shell
+curl --request PUT \
+  --url https://api.netbird.io/api/integrations/google-idp/<ID> \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <PAT>' \
+  --header 'Content-Type: application/json' \
+  --data '{
+	"serviceAccountKey": "<SERVICE_ACCOUNT_KEY>",
+	"syncInterval": 300,
+	"enabled": false
+}'
+```
+
+Response
+```json
+{
+  "id": <ID>,
+  "customerId": "<CUSTOMER_ID>",
+  "syncInterval": 300,
+  "enabled": false
+}
+```
+
+### Delete Integration
+Request
+```shell
+curl --request DELETE \
+  --url https://api.netbird.io/api/integrations/google-idp/<ID> \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <PAT>'
+```
+Response
+```json
+{}
+```
+
+### Get Integration sync logs
+Request
+```shell
+curl --request GET \
+  --url https://api.netbird.io/api/integrations/google-idp/<ID>/logs \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <PAT>'
+```
+Response
+```json
+[
+  {
+    "id": <ID>,
+    "level": "info",
+    "timestamp": "timestamp UTC",
+    "message": "message"
+  }
+]
+```
+
+
+## Azure Endpoints
```

Comment:
Similar as the comment to the Google endpoints section.
Is there any specific name for Azure? maybe also including a link to the landing page of the system so that people know exactly what we are describing here.